### PR TITLE
[FLINK-17416][e2e][k8s] Use fixed v1.16.9 because fabric8 kubernetes-client could not work with higher version under jdk 8u252

### PR DESCRIPTION
**Pull request by wangyang0918**
## What is the purpose of the change

Use fixed v1.16.9 because fabric8 kubernetes-client could not work with higher version under jdk 8u252.

* For jdk 8u252, the fabric8 kubernetes-client could only work on kubernetes v1.16 and lower versions.
* For other jdk versions(e.g. 8u242, jdk11), i am not aware of the same issues. It seems that native K8s integration works well. The user in the ML have solved the problem after upgrading to jdk11.

## Brief change log

* Update kubernetes version of minikube
* Enable K8s related tests

## Verifying this change

* Covered by existing tests

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by wangyang0918** [comment]
cc @rmetzger

File: flink-end-to-end-tests/test-scripts/common_kubernetes.sh [hunk 1]
```diff
@@ -75,7 +75,8 @@ function start_kubernetes_if_not_running {
         # here.
         # Similarly, the kubelets are marking themself as "low disk space",
         # causing Flink to avoid this node (again, failing the test)
-        sudo CHANGE_MINIKUBE_NONE_USER=true minikube start --vm-driver=none \
+        # Use fixed version v1.16.9 because fabric8 kubernetes-client could not work with higher version under jdk 8u252
+        sudo CHANGE_MINIKUBE_NONE_USER=true minikube start --kubernetes-version v1.16.9 --vm-driver=none \
```
Comment (rmetzger) [inline flink-end-to-end-tests/test-scripts/common_kubernetes.sh:79]: which k8s version is the e2e test usually using?

Comment (wangyang0918) [reply flink-end-to-end-tests/test-scripts/common_kubernetes.sh:79]: Currently, we are using minikube `v1.8.2`, the default corresponding Kubernetes version is `v1.17` by default.

Comment (rmetzger) [reply flink-end-to-end-tests/test-scripts/common_kubernetes.sh:79]: Okay, I see. Thx!

Does it make sense to run the K8s tests with multiple K8s versions to better detect such issues in the future? We test against the latest version (1.18), oldest maintained (1.16)?

Comment (wangyang0918) [reply flink-end-to-end-tests/test-scripts/common_kubernetes.sh:79]: This is a great idea. I will create a ticket to track this.

**Comment by rmetzger** [comment]
CI has passed. Merging.

**Pull request merged**